From 79c225a61b496f9c81ce9dd31fb61155fc59b1cc Mon Sep 17 00:00:00 2001
From: Kenny Kessler <kenny.kessler@motorola.com>
Date: Wed, 21 Oct 2015 15:02:34 -0500
Subject: [PATCH 413/959] IKSWM-8062: power: qpnp-smbcharger-mmi: Fix HVDCP

Fix the Charge Rate detection so that it supports
PMI8950 variants with SMBLITE register sets.

Change-Id: Iab9a131624101c4b2f8f316ec04117b38db4af49
Signed-off-by: Kenny Kessler <kenny.kessler@motorola.com>
Reviewed-on: http://gerrit.mot.com/793366
SME-Granted: SME Approvals Granted
SLTApproved: Slta Waiver <sltawvr@motorola.com>
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Jason Knopsnyder <jasonk@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
---
 drivers/power/qpnp-smbcharger-mmi.c | 18 ++++++++++++++----
 1 file changed, 14 insertions(+), 4 deletions(-)

diff --git a/drivers/power/qpnp-smbcharger-mmi.c b/drivers/power/qpnp-smbcharger-mmi.c
index 9912ff2c682..d6d1f378272 100644
--- a/drivers/power/qpnp-smbcharger-mmi.c
+++ b/drivers/power/qpnp-smbcharger-mmi.c
@@ -3563,7 +3563,7 @@ static void smbchg_aicl_deglitch_wa_check(struct smbchg_chip *chip)
 {
 	union power_supply_propval prop = {0,};
 	int rc;
-	u8 reg;
+	u8 reg, hvdcp_sel;
 	bool low_volt_chgr = true;
 
 	if (!(chip->wa_flags & SMBCHG_AICL_DEGLITCH_WA))
@@ -3578,6 +3578,11 @@ static void smbchg_aicl_deglitch_wa_check(struct smbchg_chip *chip)
 	if (!chip->bms_psy)
 		return;
 
+	if (chip->schg_version == QPNP_SCHG_LITE)
+		hvdcp_sel = SCHG_LITE_USBIN_HVDCP_SEL_BIT;
+	else
+		hvdcp_sel = USBIN_HVDCP_SEL_BIT;
+
 	if (is_usb_present(chip)) {
 		rc = smbchg_read(chip, &reg,
 				chip->usb_chgpth_base + USBIN_HVDCP_STS, 1);
@@ -3585,7 +3590,7 @@ static void smbchg_aicl_deglitch_wa_check(struct smbchg_chip *chip)
 			pr_err("Couldn't read hvdcp status rc = %d\n", rc);
 			return;
 		}
-		if (reg & USBIN_HVDCP_SEL_BIT)
+		if (reg & hvdcp_sel)
 			low_volt_chgr = false;
 	} else if (is_dc_present(chip)) {
 		if (chip->dc_psy_type == POWER_SUPPLY_TYPE_WIPOWER)
@@ -3623,7 +3628,7 @@ static void smbchg_aicl_deglitch_wa_check(struct smbchg_chip *chip)
 static bool smbchg_hvdcp_det_check(struct smbchg_chip *chip)
 {
 	int rc;
-	u8 reg;
+	u8 reg, hvdcp_sel;
 
 	rc = smbchg_read(chip, &reg,
 			chip->usb_chgpth_base + USBIN_HVDCP_STS, 1);
@@ -3632,11 +3637,16 @@ static bool smbchg_hvdcp_det_check(struct smbchg_chip *chip)
 		return false;
 	}
 
+	if (chip->schg_version == QPNP_SCHG_LITE)
+		hvdcp_sel = SCHG_LITE_USBIN_HVDCP_SEL_BIT;
+	else
+		hvdcp_sel = USBIN_HVDCP_SEL_BIT;
+
 	/*
 	 * If a valid HVDCP is detected, notify true only
 	 * if USB is still present.
 	 */
-	if ((reg & USBIN_HVDCP_SEL_BIT) && is_usb_present(chip))
+	if ((reg & hvdcp_sel) && is_usb_present(chip))
 		return true;
 
 	return false;
-- 
2.11.0

